classes/Http/Action/Admin/Talk/IndexAction.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Action\Admin\Talk;
+
+use OpenCFP\Domain\Services;
+use OpenCFP\Domain\Talk;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation;
+
+final class IndexAction
+{
+    /**
+     * @var Services\Authentication
+     */
+    private $authentication;
+
+    /**
+     * @var Talk\TalkFilter
+     */
+    private $talkFilter;
+
+    public function __construct(Services\Authentication $authentication, Talk\TalkFilter $talkFilter)
+    {
+        $this->authentication = $authentication;
+        $this->talkFilter     = $talkFilter;
+    }
+
+    /**
+     * @Template("admin/talks/index.twig")
+     *
+     * @param HttpFoundation\Request $request
+     *
+     * @return array
+     */
+    public function __invoke(HttpFoundation\Request $request): array
+    {
+        $adminUserId = $this->authentication->user()->getId();
+
+        $options = [
+            'order_by' => $request->get('order_by'),
+            'sort'     => $request->get('sort'),
+        ];
+
+        $formattedTalks = $this->talkFilter->getTalks(
+            $adminUserId,
+            $request->get('filter'),
+            $options
+        );
+
+        $perPage = (int) $request->get('per_page') ?: 20;
+
+        $pagerfanta = new Services\Pagination($formattedTalks, $perPage);
```

Comment:
Because we create dependencies here rather than injecting them, we cannot write any unit tests.